RecoTracker/LSTCore/src/alpaka/Quadruplet.h

```diff
@@ -0,0 +1,2404 @@
+                  float phi =
+                      mdsInGPU.anchorPhi[segmentsInGPU.mdIndices[2 * tripletsInGPU.segmentIndices[2 * innerTripletIndex]]];
+                  float eta =
+                      mdsInGPU.anchorEta[segmentsInGPU.mdIndices[2 * tripletsInGPU.segmentIndices[2 * innerTripletIndex]]];
+                  float pt = (innerRadius + outerRadius) * lst::k2Rinv1GeVf;
+                  addQuadrupletToMemory(tripletsInGPU,
+                                        quadrupletsInGPU,
+                                        innerTripletIndex,
+                                        outerTripletIndex,
+                                        lowerModule1,
+                                        lowerModule2,
+                                        lowerModule3,
+                                        lowerModule4,
+                                        innerRadius,
+                                        outerRadius,
+                                        pt,
+                                        eta,
+                                        phi,
+                                        // scores,
+                                        layer,
+                                        quadrupletIndex,
+                                        rzChiSquared,
+                                        dBeta);
+
+                  // tripletsInGPU.partOfT4[quadrupletsInGPU.tripletIndices[2 * quadrupletIndex]] = true;
+                  // tripletsInGPU.partOfT4[quadrupletsInGPU.tripletIndices[2 * quadrupletIndex + 1]] = true;
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  struct createEligibleModulesListForQuadrupletsGPU {
+    template <typename TAcc>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc,
+                                  lst::Modules modulesInGPU,
+                                  lst::Triplets tripletsInGPU,
+                                  lst::ObjectRanges rangesInGPU,
+                                  const float ptCut) const {
+      auto const globalThreadIdx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+      auto const gridThreadExtent = alpaka::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+
+      // Initialize variables in shared memory and set to 0
+      int& nEligibleT4Modulesx = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+      int& nTotalQuadrupletsx = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+      if (cms::alpakatools::once_per_block(acc)) {
+        nTotalQuadrupletsx = 0;
+        nEligibleT4Modulesx = 0;
+      }
+      alpaka::syncBlockThreads(acc);
+
+      // Occupancy matrix for 0.8 GeV pT Cut
+      constexpr int p08_occupancy_matrix[4][4] = {
+          {5000, 5000, 5000, 5000},  // category 0
+          {5000, 5000, 5000, 5000},          // category 1
+          {5000, 5000, 5000, 5000},          // category 2
+          {5000, 5000, 5000, 5000}       // category 3
+      };
```

Comment:
I suspect the results shown today were truncated in some cases.
I suggest to merge #148 to moderate the memory blowup and increase the above by x10 and redo the inclusive T3 input test

Reply:
You will need to add similar logic to the Quadruplet kernel here if you want to increase these 10x btw, since I only incorporated it into the existing objects. Should be only a few lines though that you need to add. See my change for the T5 occupancies as a reference.